utils/src/parallel_restore.c

```diff
@@ -1136,6 +1136,7 @@ static spr_err_t prepare_alloc_builders(struct scoutfs_parallel_restore_writer *
 	dprintf("starting prepare with start %llu len %llu\n", wri->meta_start, wri->meta_len);
 
 	skip = ALLOC_BLOCKS + (SCOUTFS_ALLOC_LIST_MAX_BLOCKS * 2);
+	skip += le64_to_cpu(wri->super.total_meta_blocks) / 10;
```

Comment:
"skip" here is the number of blocks that will be used to store allocation structures.  It reflects the number of blocks written by this routine.  By adding the reservation to skip here you're saying that it won't be available as a free record somewhere (either in list block entries or extent items).  Look at the very next thing done with the skip value, it's subtracted from a free extent that's recorded.   So we want to add it to a free extent somewhere.  The next problem is that you can't just add to any old length.  You need to think about the specific extent value that you're removing with the initial / 10 reservation.  It's the end of the block space.  To return that reservation you have to specifically extend the final extent or create a new one.

Reply:
What would the benefit be for adding a new extent onto the end vs extending the final extent?

Reply:
If you extend an existing extent you don't have to create a new one, but you have to identify the extent to extend.

Reply:
Could you elaborate a bit more on identifying an extent, I see that there is some math identifying its bounds, start and len, would I just be able to calculate the end of the reserved space and add onto it?

Reply:
Could you elaborate a bit more on identifying an extent, I see there's some math there identifying its bounds, couldn't I just calculate the end of the free extent and add a percentage of blocks into the extent?